srml/support/procedural/src/storage/transformation.rs

```diff
@@ -281,48 +291,79 @@ fn decl_store_extra_genesis(
 				}).unwrap_or_else(|| quote!( Default::default() ));
 
 			config_field_default.extend(quote!( #ident: #fielddefault, ));
+
+			opt_build
+				.map(Some)
+				.unwrap_or_else(||
+					Some(quote!( (|config: &Self| config.#ident.clone()) ))
+				)
 		} else {
-			opt_build = build.inner.as_ref().map(|b| &b.expr.content).map(|b| quote!( #b ));
-		}
+			opt_build
+		};
 
 		let typ = type_infos.typ;
-		if let Some(builder) = opt_build {
-			is_trait_needed = true;
+		if let Some(builder) = builder {
+			assimilate_require_generic = true;
```

Comment:
is that an error ? then you reset it to true again for some value in the match

Reply:
Hmm yeah.